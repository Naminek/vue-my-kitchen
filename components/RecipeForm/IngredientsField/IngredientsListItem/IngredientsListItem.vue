<template>
  <div>
    <v-row dense>
      <v-col
        class="flex-shrink-1 flex-grow-0"
        cols="11"
        sm="4"
        md="5"
      >
        <v-autocomplete
          :value="localItem.ingredient_id"
          label="Item"
          outlined
          dense
          hide-details="auto"
        />
      </v-col>
      <v-col
        class="flex-shrink-1 flex-grow-0"
        cols="7"
        sm="4"
        md="5"
      >
        <v-text-field
          :value="localItem.amount"
          label="Amount"
          outlined
          dense
          hide-details="auto"
        />
      </v-col>
      <v-col
        class="flex-shrink-1 flex-grow-0"
        cols="4"
        sm="1"
        style="min-width: 73px"
      >
        <v-select
          :value="localItem.group"
          label="Group"
          :items="groupItems"
          outlined
          dense
          hide-details="auto"
        />
      </v-col>
    </v-row>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from '@nuxtjs/composition-api'
import SmoothReflow from '~/components/Common/SmoothReflow'
import { Group } from '~/composables/enum'
import { usePropWatch } from '~/composables/props'
import IngredientItem from '~/composables/type'

export default defineComponent({
  name: 'IngredientsListItem',
  components: { SmoothReflow },
  props: {
    item: {
      type: Object as PropType<IngredientItem>,
      default: () => {}
    }
  },
  setup (props) {
    const propsWatch = usePropWatch(props)
    const localItem = propsWatch('item')

    const groupItems = Object.values(Group)

    return { localItem, groupItems }
  }
})
</script>
