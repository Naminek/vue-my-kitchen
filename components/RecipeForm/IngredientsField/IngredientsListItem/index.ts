import IngredientsListItem from './IngredientsListItem.vue'
export default IngredientsListItem
