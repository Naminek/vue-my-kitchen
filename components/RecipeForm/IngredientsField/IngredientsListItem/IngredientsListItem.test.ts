import { getFixedIngredientItems } from '~/test/fakes/ingredientitem'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'
import IngredientsListItem from '.'

const localVue = createLocalVue()

const ingredientItem = getFixedIngredientItems(1)[0]

describe('IngredientsListItem', () => {
  const defaultProps = {
    item: ingredientItem
  }

  beforeEach(() => {
    jest.resetAllMocks()
  })

  function mountFunc (options = {}) {
    return mount(IngredientsListItem, {
      localVue,
      vuetify: new Vuetify(),
      propsData: defaultProps,
      ...options
    })
  }

  it('should match its snapshot', () => {
    const wrapper = mountFunc()
    expect(wrapper).toMatchSnapshot()
  })
})
