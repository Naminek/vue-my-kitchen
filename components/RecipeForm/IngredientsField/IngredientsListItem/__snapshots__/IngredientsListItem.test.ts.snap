// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsListItem should match its snapshot 1`] = `
<div>
  <div class="row row--dense">
    <div class="flex-shrink-1 flex-grow-0 col-sm-4 col-md-5 col-11">
      <div class="v-input v-input--hide-details v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
        <div class="v-input__control">
          <div aria-expanded="false" aria-haspopup="listbox" aria-owns="list-2" class="v-input__slot" role="combobox">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;">
                <span class="notranslate">​</span>
              </legend>
            </fieldset>
            <div class="v-select__slot">
              <label class="v-label theme--light" for="input-2" style="left: 0px; position: absolute;">Item</label>
              <input autocomplete="off" id="input-2" type="text">
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append">
                  <i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i>
                </div>
              </div>
              <input type="hidden" value="1">
            </div>
            <div class="v-menu">
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="flex-shrink-1 flex-grow-0 col-sm-4 col-md-5 col-7">
      <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;">
                <span class="notranslate">​</span>
              </legend>
            </fieldset>
            <div class="v-text-field__slot">
              <label class="v-label v-label--active theme--light" for="input-5" style="left: 0px; position: absolute;">Amount</label>
              <input id="input-5" type="text">
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="flex-shrink-1 flex-grow-0 col-sm-1 col-4" style="min-width: 73px;">
      <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select">
        <div class="v-input__control">
          <div aria-expanded="false" aria-haspopup="listbox" aria-owns="list-6" class="v-input__slot" role="button">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;">
                <span class="notranslate">​</span>
              </legend>
            </fieldset>
            <div class="v-select__slot">
              <label class="v-label v-label--active theme--light" for="input-6" style="left: 0px; position: absolute;">Group</label>
              <div class="v-select__selections">
                <div class="v-select__selection v-select__selection--comma">2</div>
                <input aria-readonly="false" autocomplete="off" id="input-6" readonly="readonly" type="text">
              </div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--append">
                  <i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i>
                </div>
              </div>
              <input type="hidden" value="2">
            </div>
            <div class="v-menu">
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
